From ac2279a50aee4c8c5fabb645d185ad6c9e61d553 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Thu, 15 Aug 2013 15:07:57 -0400
Subject: [PATCH] units: make fsck units remain after exit

Without this, fsck would be re-run if any other service which pulls
in a target requiring one of the mounts was started after fsck was done
but before the initial transaction was done.

https://bugs.freedesktop.org/show_bug.cgi?id=66784
---
 units/systemd-fsck-root.service.in | 4 ++--
 units/systemd-fsck@.service.in     | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/units/systemd-fsck-root.service.in b/units/systemd-fsck-root.service.in
index 563129b..4388314 100644
--- a/units/systemd-fsck-root.service.in
+++ b/units/systemd-fsck-root.service.in
@@ -7,7 +7,7 @@
 
 [Unit]
 Description=File System Check on Root Device
-Documentation=man:systemd-fsck@.service(8)
+Documentation=man:systemd-fsck-root.service(8)
 DefaultDependencies=no
 After=systemd-readahead-collect.service systemd-readahead-replay.service
 Before=local-fs.target shutdown.target
@@ -16,7 +16,7 @@ ConditionPathIsReadWrite=!/
 
 [Service]
 Type=oneshot
-RemainAfterExit=no
+RemainAfterExit=yes
 ExecStart=@rootlibexecdir@/systemd-fsck
 StandardOutput=journal+console
 FsckPassNo=1
diff --git a/units/systemd-fsck@.service.in b/units/systemd-fsck@.service.in
index b3c71eb..e229cdc 100644
--- a/units/systemd-fsck@.service.in
+++ b/units/systemd-fsck@.service.in
@@ -15,7 +15,7 @@ Before=shutdown.target
 
 [Service]
 Type=oneshot
-RemainAfterExit=no
+RemainAfterExit=yes
 ExecStart=@rootlibexecdir@/systemd-fsck %f
 StandardOutput=journal+console
 TimeoutSec=0
